.github/workflows/ci.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+      - '*'
```

Comment:
[Q] Should it run on every branch / PR ?

Reply:
Let's do:

```
on:
  push:
    branches: [main]
  pull_request:
    - '*'
```

ie. any pull request (which includes any push to a branch in an open pull request), and any push to main, but not pushes to branches that aren't main and aren't in a pull request.

Reply:
Sorry, I typod, you wouldn't have both `[ push, pull_request ]` and the options below. Should be this:

```
on:
  push:
    branches: [main]
  pull_request:
    branches: '*'
```